src/DataSourceProvider.vue

```diff
@@ -151,24 +151,26 @@ export default {
 
       this.loadDataSources(this.widgetData.appId);
     },
-    addAppToExistginRule() {
+    addAppToExistingRule() {
       this.selectedDataSource.accessRules.forEach(dataSourceRule => {
-        if (dataSourceRule.appId && !dataSourceRule.appId.includes(this.widgetData.appId)) {
-          if (!_.difference(this.missingAccessTypes, dataSourceRule.type).length) {
-            return dataSourceRule.appId.push(this.widgetData.appId);
-          }
+        if (dataSourceRule.appId && dataSourceRule.appId.includes(this.widgetData.appId)) {
```

Comment:
what if the dataSourceRule does not have an appId?

Reply:
@squallstar Yes i checked the appId and it can be `null or []`, so i changed the condition to fix it

Reply:
@squallstar, i tested the scenario if we does not have an appId and it can be `null or []`, so i changed the if condition to resolve this problem

Reply:
@AndrRyaz what happens when `appId` is set to an empty array? That surely is a "truthy" value which would pass the first condition, is that expected?

Reply:
@tonytlwu FYI it looked like this wasn't tested as a scenario by the developer.

Reply:
@squallstar you are right, I added a condition if we have an empty array

Reply:
Next time please test your code changes for all use cases that apply, cc @sofiiakvasnevska